### PR TITLE
cdc: fix server etcd may nil by create a new local client.

### PR DESCRIPTION
**Pull request by JinLingChristopher**
### What problem does this PR solve? 

server etcdclient may be nil, fix it by create one.

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release note

- No release note

### CODE REVIEW

File: cdc/server.go [hunk 1]
```diff
@@ -392,15 +392,30 @@ func (s *Server) setUpDataDir(ctx context.Context) error {
 		return nil
 	}
 
+	cli := s.etcdClient
```
Comment (amyangfei) [inline cdc/server.go:395]: The new replicaImpl creates an `etcdClient` uses it when creating a new capture. The `etcdClient` won't be closed or cleaned up until the cdc server exits.

While the old replicaImpl creates a new `etcdClient` every time a server `run` is called. https://github.com/pingcap/ticdc/blob/9972085455542df023aaecb306248936a7fdcf8b/cdc/server.go#L300-L305

Maybe the new replicaImpl way is better. But I **don't think we need to** change this logic in this PR.

File: cdc/server.go [hunk 2]
```diff
@@ -392,15 +392,30 @@ func (s *Server) setUpDataDir(ctx context.Context) error {
...
+	if cli == nil {
```
Comment (amyangfei) [inline cdc/server.go:396]: Better to add an unit test to cover this condition.

File: cdc/server.go (hunk 1 above)
Comment (ben1009) [inline cdc/server.go:395]: IMHO, it is fine for a quick fix, but could u add some comments explain why the fix is needed, and ut to cover the case in the issue, thx

**Comment by ben1009** [comment]
besides the pr, do we have compatible tests to cover cases like this, thx

**Review by amyangfei — APPROVED** [review]

**Review by leoppro — APPROVED** [review]

**Pull request merged**